optimade/validator/validator.py

```diff
@@ -544,3 +593,18 @@ def test_mandatory_query_syntax(self, endpoint, endpoint_queries):
         valid_queries = [f"{endpoint}?filter={query}" for query in endpoint_queries]
         for query in valid_queries:
             self.get_endpoint(query)
+
+    def test_optional_query_syntax(self, endpoint, endpoint_queries):
```

Comment:
Why not just create a `test_query_syntax` adding the `optional` flag instead of having two almost identical methods?

Reply:
Or just an additional "helper" method `_test_query_syntax` that these methods can call, passing a "hardcoded" `optional` flag, if you wish to keep distinct methods for this.